Application/Queries/CompensationsQuery.cs

```diff
@@ -9,27 +9,21 @@ public class CompensationsQuery : ICompensationsQuery
 {
     private readonly CompensationsDbContext _context;
 
-    public CompensationsQuery(CompensationsDbContext context)
+    private readonly ITenantCompensationsQuery _tenantCompensationsQuery;
+
+    public CompensationsQuery(CompensationsDbContext context, ITenantCompensationsQuery tenantCompensationsQuery)
```

Comment:
You should keep only CompensationsQuery and it knows about tenants. Look at the comment about TenantCompensationsQuery.